src/whiteflag/wf_core/segment.rs

```diff
@@ -34,7 +34,7 @@ impl MessageSegment {
                 throw new WfCoreException("Field " + field.debugInfo() + " already set or array item " + index + " contains invalid data: " + data[index], null);
             } */
             let value = &data[index];
-            field.set(value.as_ref());
+            field.set(value.as_ref()).unwrap();
```

Comment:
@Romulus10 I haven't fleshed out the error handling -- I began adding it to one or two functions but decided it could wait -- more functions need their outputs changed to handle errors and I can also add unit tests to make sure the errors are being triggered properly -- but I didn't think we needed to worry about that for M3